### PR TITLE
add 2 targets: "tex" and "lint" to Makefile

### PR DESCRIPTION
**Pull request by EN-Chou**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -63,7 +63,7 @@ ALL_EPS := $(patsubst $(SCHEMATIC_DIR)/%.tex,$(EPS_DIR)/%.eps,$(ALL_TEX)) \
 	$(patsubst $(SCHEMATIC_DIR)/%.py,$(EPS_DIR)/%.eps,$(ALL_SCHPY))
 
 .PHONY: default
-default: cese mesh projection
+default: cese mesh projection
```
Comment (tai271828) [inline Makefile:66]: Extra trailing space.

**Review by tai271828 — COMMENTED** [review]
Minor comment inline.

Looks good to me overall. However, I have not tried chktex before. Let me try it later on and update my feedback here.

File: Makefile [hunk 2]
```diff
@@ -147,6 +147,14 @@ projection_ho: projection.pdf
 	mkdir -p $(HANDOVER_DIR)
 	cp -f $< $(HANDOVER_FN)
 
+
+.PHONY: tex
+tex: 
+	sudo apt install chktex
+
+.PHONY: lint
+lint: tex
```
Comment (tai271828) [inline Makefile:156]: You only install `chktex`, but you do not use `chktex` to do something. For example, you may add lines like `chetex projection.tex` to ask for `chktext` to check `projection.tex` for you.

After completing this, you need to edit `.github/workflows/build_doc.yml` to append lines like `make lint` to make the lint checking automatic via Github Action.

**Review by tai271828 — CHANGES_REQUESTED** [review]
It will be great if we make the Makefile more comprehensive. See the inline comments :)

File: Makefile [hunk 3]
```diff
@@ -147,6 +147,14 @@ projection_ho: projection.pdf
 	mkdir -p $(HANDOVER_DIR)
 	cp -f $< $(HANDOVER_FN)
 
+
+.PHONY: tex
+tex: 
+	sudo apt install chktex
```
Comment (yungyuc) [inline Makefile:153]: Install or sudo are to manage the system.  Makefile should not do either.

**Comment by yungyuc** [comment]
This PR is linked to #7

**Comment by tai271828** [comment]
I think this pull request is ready to merge. @EN-Chou do you prefer to land the pull request first, or you want to work on `.github/workflows/build_doc.yml` as the same time in this thread of pull request?

**Comment by EN-Chou** [comment]
> I think this pull request is ready to merge. @EN-Chou do you prefer to land the pull request first, or you want to work on `.github/workflows/build_doc.yml` as the same time in this thread of pull request?

Is there convention for this?

**Comment by EN-Chou** [comment]
> I think this pull request is ready to merge. @EN-Chou do you prefer to land the pull request first, or you want to work on `.github/workflows/build_doc.yml` as the same time in this thread of pull request?

What is the convention for this problem?

**Comment by tai271828** [comment]
> > I think this pull request is ready to merge. @EN-Chou do you prefer to land the pull request first, or you want to work on `.github/workflows/build_doc.yml` as the same time in this thread of pull request?
> 
> What is the convention for this problem?

Both works. If you don't have preference, let's merge? @yungyuc what do you think?

**Comment by yungyuc** [comment]
I think it makes sense to include the CI execution in this PR.

**Comment by EN-Chou** [comment]
Okay, let me check out .github/workflows/build_doc.yml next, I'll try to make a commit today.

**Comment by tai271828** [comment]
> Okay, let me check out .github/workflows/build_doc.yml next, I'll try to make a commit today.

I made a pull request for your reference https://github.com/EN-Chou/mmnote/pull/2 to help you move forward, and you can see the code difference https://github.com/EN-Chou/mmnote/pull/2/files

The corresponding result could be found here:
- https://github.com/tai271828/mmnote/actions/runs/3958461418 for commit b8f014c
- https://github.com/tai271828/mmnote/actions/runs/3958461418 for commit 0a0c1c7

File: .github/workflows/build_doc.yml [hunk 4]
```diff
@@ -24,10 +24,12 @@ jobs:
       run: |
         sudo apt-get -qqy update
         sudo apt-get -qy install texlive-latex-extra texlive-pstricks texlive-science
+        sudo apt install chktex
 
     - name: make
       run: |
         make
+        make lint
```
Comment (tai271828) [inline .github/workflows/build_doc.yml:32]: I would suggest you to separate lint checking from generating pdf. For example: https://github.com/solvcon/mmnote/commit/b8f014c171e146691358ae49708f41e6f5749097

**Comment by yungyuc** [comment]
After adding `make lint` to CI, it needs to pass before the PR can be merged.  The test currently fails: https://github.com/solvcon/mmnote/actions/runs/3960089831/jobs/6783753529

**Comment by EN-Chou** [comment]
> After adding `make lint` to CI, it needs to pass before the PR can be merged.  The test currently fails: https://github.com/solvcon/mmnote/actions/runs/3960089831/jobs/6783753529

Okay, I'll try to find out what the problem is today.

**Review by tai271828 — CHANGES_REQUESTED** [review]
Hi, I thought I have put my comments earlier. They are:
1. put `make lint` in another job name rather then `name: make`
2. figure out the chktex error and make it pass

**Comment by EN-Chou** [comment]
Hi @yungyuc and @tai271828, I've turned off warning 2, 8 and 24 by editing `.chktexrc` via tracing hyperlink to the right in section "7 Explanation of error messages" in the manual (https://www.nongnu.org/chktex/ChkTeX.pdf#page=28). 

Before (there are warning 2, 3, 8, 13 and 24):
```, 
~/git-repo/mmnote$ chktex ustmesh.tex 
ChkTeX v1.7.6 - Copyright 1995-96 Jens T. Berger Thielemann.
Compiled with PCRE regex support.
Warning 8 in ustmesh.tex line 3: Wrong length of dash may have been used.
\RequirePackage[2020-02-02]{latexrelease}  
                    ^
Warning 8 in ustmesh.tex line 3: Wrong length of dash may have been used.
\RequirePackage[2020-02-02]{latexrelease}  
                       ^
Warning 13 in ustmesh.tex line 75: Intersentence spacing (`\@') should perhaps be used.
solution element (CESE) method and SOLVCON.  
                                          ^
Warning 24 in ustmesh.tex line 81: Delete this space to maintain correct pagereferences.
\label{s:concept}  
^
Warning 24 in ustmesh.tex line 109: Delete this space to maintain correct pagereferences.
\label{s:storage}  
^
Warning 2 in ustmesh.tex line 128: Non-breaking space (`~') should have been used.
are are listed in Table~\ref{t:elm:meta}, \ref{t:subent1d}, \ref{t:subent2d},  
                                         ^
...
Warning 3 in ustmesh.tex line 439: You should enclose the previous parenthesis with `{}'.
    \sum\limits_{i=0}^{M-1}\left(\rho^{(i)}\right)^{\alpha}\mathbf{g}^{(i)}  
                                                 ^
No errors printed; 33 warnings printed; No user suppressed warnings; No line suppressed warnings.
See the manual for how to suppress some or all of these warnings/errors.
```

After (there are warning 3 and 13; 2, 8 and 24 were turned-off):
```
~/git-repo/mmnote$ chktex ustmesh.tex 
ChkTeX v1.7.6 - Copyright 1995-96 Jens T. Berger Thielemann.
Compiled with PCRE regex support.
Warning 13 in ustmesh.tex line 75: Intersentence spacing (`\@') should perhaps be used.
solution element (CESE) method and SOLVCON.  
                                          ^
Warning 13 in ustmesh.tex line 290: Intersentence spacing (`\@') should perhaps be used.
Node definition of one- and two-dimensional mesh elements in SOLVCON:  
                                                                    ^
Warning 13 in ustmesh.tex line 328: Intersentence spacing (`\@') should perhaps be used.
Node definitions of three-dimensional mesh elements in SOLVCON:  
                                                              ^
Warning 13 in ustmesh.tex line 393: Intersentence spacing (`\@') should perhaps be used.
vertices of a FGE.  The coordinates define $N$ \textit{displacement vectors}  
                 ^
Warning 3 in ustmesh.tex line 439: You should enclose the previous parenthesis with `{}'.
    \sum\limits_{i=0}^{M-1}\left(\rho^{(i)}\right)^{\alpha}\mathbf{g}^{(i)}  
                                                 ^
Warning 3 in ustmesh.tex line 441: You should enclose the previous parenthesis with `{}'.
    \sum\limits_{i=0}^{M-1}\left(\rho^{(i)}\right)^{\alpha}  
                                                 ^
No errors printed; 6 warnings printed; No user suppressed warnings; No line suppressed warnings.
See the manual for how to suppress some or all of these warnings/errors.
```

Yet, I'm currently stuck while trying to turn off warning 3, 10, 17, 25, 36 and 37, where they do not have hyperlink to the right. Can you please help me with this problem?

**Comment by yungyuc** [comment]
Please see https://www.nongnu.org/chktex/ChkTeX.pdf#page=10 for the per line suppression.

The warnings 3, 10, 17, 25, 36, and 37 look to be related to curly braces and/or parentheses.  It makes sense to always check for them and perhaps that is why chktex does not provide options.  You may use the per line suppression to turn them off.

**Comment by EN-Chou** [comment]
Thank you @yungyuc , I did per file suppression instead in(https://github.com/solvcon/mmnote/commit/ef3b64e9ff755e1601717a532ae61aa8a92c0cae).

**Review by yungyuc — APPROVED** [review]
LGTM

**Comment by yungyuc** [comment]
It works beautifully: https://github.com/solvcon/mmnote/actions/runs/4012601016/jobs/6891165262#step:5:8

**Pull request merged**